### ISSUE TITLE
Is wp.customizer() currently broken?

### ISSUE THREAD

**Issue by kmob2** [reduxframework/redux-framework#2577]
I am using the following javascript for some live previews inside the customizer.

wp.customize( 'test_theme[fluid_design]', function( value ) {
	value.bind( function( newval ) {
		console.log(newval);
		$('.site-container').css('max-width', '100%' );
		} );
} );
This code snipped is part of the theme-customizer.js to get live previews (https://codex.wordpress.org/Theme_Customization_API)
 
I updated from redux 3.5.6.1 to 3.5.7.2 and now it's not working anymore. Nothing inside the wp.customize value.bind is being called. No error message either.
When I switch back to the old version, it is working again. Looking at the change log, I see some changes were made in the customizer. Maybe it broke the wp.customize function?

**Comment by kmob2** [comment]
Any update or ETA on this? I am testing now with 3.5.7.9 and still doesn't work. Seems quite a big issue that the (custom) customizer live preview don't work, this kind of renders the entire work for the customizer useless. In simple words, redux customizer can't currently be used to create live previews using PostMessage.
I am happy to provide more details or examples if it's not clear what is broken.

**Comment by kprovance** [comment]
I don't work with the customizer, that's @Dovy's area, and he'll reply when he is able, since he works full time and has a family.  I'm sorry it's taking so long.

**Comment by kmob2** [comment]
Sure thing, I understand and appreciate all the work, thus I have donated in the past. I'm currently looking into making another donation to get the ball rolling on this. If this finally works, I'm all set to use my theme in a live environment :)

**Comment by contempoinc** [comment]
Any update on this?

**Comment by kprovance** [comment]
Believe me, I wish I understood the whole customizer API well enough to remotely understand the issue.  And Dovy is slammed with his day job, for weeks to come if I heard him correctly.  I hate to ask the obvious, but have you looked at the customer JS in the Redux core to see if you can spot the obvious flaw?  You seem to understand the concepts WAY better than I.

**Comment by kmob2** [comment]
I am also checking daily to see if this got fixed. All I know is that it broke with the update on 26th August (v. 3.5.7),
The following files were changed and in one or all of them lies the issue (don't ask me where). Github has nice code highlighting and comparison with previous versions. Maybe someone familiar with the customizer might know where to start. For me it will be a trial and error run.

/redux-framework/ReduxCore/inc/extensions/customizer/extension_customizer.php
/redux-framework/ReduxCore/inc/extensions/customizer/inc/customizer_control.php
/redux-framework/ReduxCore/inc/extensions/customizer/inc/customizer_panel.php
/redux-framework/ReduxCore/inc/extensions/customizer/inc/customizer_section.php

I think the biggest issue is, that this feature was working and then it broke and since then the customizer is basically rendered useless. Considering we all donated in the backer campaign only to have the customizer implemented by ~October, I was hoping this issue would get some attention. But at the same time I fully understand all the free work you guys are putting into this and I appreciate every free minute you guys invest into this.
Sometimes I wish you guys would commercialize redux and charge i.e $100/year for this, so that you 2 get at least something from it. Trust me, I sign up in a heart beat if I would know you would be able to work more on redux.

**Comment by kprovance** [comment]
<blockquote>
Sometimes I wish you guys would commercialize redux and charge i.e $100/year for this, so that you 2 get at least something from it. Trust me, I sign up in a heart beat if I would know you would be able to work more on redux.
</blockquote>
We floated that idea once, a long time ago.  People balked.  Plus, a certain someone (not me, ha) promised the core would always be free.  Looking back, our typography field should have been a premium extension, well, the advanced one.

Out of curiosity, I loaded up the customizer.  The preview did render, albeit it took for freakin' ever.  Is that the issue?

**Comment by kmob2** [comment]
Too bad! I guess it's always hard to monetize from open source, especially if the script doesn't target end users. There has to be a way where you (if wanted) could work full time on this and get paid enough. After all the success is huge.

The customizer itself does load correctly. The issue is when adding your own javascript to the customizer to create live previews using the redux fields. (https://codex.wordpress.org/Theme_Customization_API)
To demonstrate this, it requires some setup and doesn't work out of the box. But basically, you create a redux field, let's say a slider. You then assign this slider to a live preview function by adding your own javascript code inside theme-customizer,js. This way I could use the slider to change the width of the website from for example 800px to 1200px by simply moving the slider inside the customizer, and get a live preview as I move the slider and seeing the websites width increase/decrease. Pretty cool feature!
Unfortunately no javascript is being executed inside wp.customize() function of the theme-customizer,js file anymore since that specific update. Also no error logs in the console.

**Comment by dovy** [comment]
Hey all, sorry about that. Honestly I had to go get a day job because Redux didn't make even close to what my family needed to survive. Been digging into BIG data (processing 260GB/day in a few months for live metrics). 

With the advanced customizer it's possible (and quite easy) to make your own Redux fields with handlers. But I will have to dig in to see why the wp.customize hook isn't running. I'm out for the next 10 days though probably. Getting ready for a big public launch and doing a ton of backend integrations.

Sorry.  :(

**Comment by Slobork** [comment]
Just a small comment on a profitability part of the Redux. Maybe you could think of switching to membership model, instead of premium addons. When you look over the WordPress market, when you are paying for the premium plugins/themes most likely you pay once for the lifetime access to the product or for the membership access. Maybe you could create some kind of poll to see would that induce more upgrades. 

I'm just building my first theme and at the moment I was able only to modestly support the Advanced Customizer addon. I cannot understand if people are making money and do not support Redux either through the donations or by buying the premium addons.

**Comment by kprovance** [comment]
<blockquote>
I cannot understand if people are making money and do not support Redux either through the donations or by buying the premium addons. 
</blockquote>
That makes three of us (Dovy and I, included).  If I had a dime for everytime someone said they would donate after they're theme was released, or they began making money, well, we'd be fully funded forever.  They conveniently forget about us, until they want help, then the thought of contributing is unthinkable.  Happened darned near *every* time.  Then they go to stack overflow looking for support.  Kinda of like taking the long way home to avoid paying the toll bridge.  That's how these folks are.  We've accepted it, but it also means diverting huge chunks of time to other activities that actually pay, which means less time for Redux.  Nature of the beast.
